test/native/helpers.js

```diff
@@ -87,16 +84,18 @@ export async function waitForStoreResolvers( fn ) {
  * @param {import('react').ReactNode} [options.component] A specific editor component to render.
  * @return {import('@testing-library/react-native').RenderAPI} A Testing Library screen.
  */
-export async function initializeEditor( props, { component = Editor } = {} ) {
+export async function initializeEditor( props, { component } = {} ) {
+	const uniqueId = uuid();
```

Comment:
I see the reason you introduced these variables makes sense because they are required by the signature of the `initializeEditor` function exported from the `@wordpress/edit-post` module.

---

Something I'm having difficulty grokking just reading this briefly is the couple use cases I found doing a quick VSCode search for `await initializeEditor( {}, { component: EditorComponent } )` with a dependency injected component. Why would we need to allow for dependency injected component in some use cases, but not all?

```
const getEditorComponent = () => {
	let EditorComponent;
	AppRegistry.registerComponent.mockImplementation(( name, componentProvider ) => { EditorComponent = componentProvider(); });
	registerGutenbergMobile();
	return EditorComponent;
};
...
const screen = await initializeEditor({ locale, capabilities }, { component: getEditorComponent() });
```

Another way of asking what I'm trying to understand is "if we suddenly want to make sure that the dispatch to the preferences store takes place when initializing an editor, but the test using this function dependency injects the Editor component, do they simply not care about the dispatch to the preferences store?" They'd have to dispatch manually in that use-case, right?

Why not do something like this?
```
const screen = render(cloneElement(internalInitializeEditor( uniqueId, postType, postId ), { initialTitle: 'test', ...props }));
```

Reply:
I was just curious, and literally only one test fails when the dependency inject `component` is not used.

**Aside:** All of the tests will likely pass with `npm run native test` without the altered code shown below since 99% of the tests passed with the `internalInitializeEditor` usage.

![image](https://user-images.githubusercontent.com/3058263/156831110-3dfdadf8-1c5f-4e87-8338-d9a0bdf3f26d.png)

Reply:
**Unaltered:**
![image](https://user-images.githubusercontent.com/3058263/156832997-c7a104de-129a-4ad4-9c78-4fd347a364b9.png)

Reply:
> Why would we need to allow for dependency injected component in some use cases, but not all?

This is only needed for the use case that tests the full editor initialization, which retrieves the editor component via the `registerGutenberg` function ([reference](https://github.com/WordPress/gutenberg/blob/bc9fbe43f9e058eab54639cff65c8d62928b5fac/packages/react-native-editor/src/index.js#L31)):
https://github.com/WordPress/gutenberg/blob/bc9fbe43f9e058eab54639cff65c8d62928b5fac/packages/react-native-editor/src/index.js#L54-L58

This behavior differs from the rest of the integration tests, which although they initialize the editor, they skip the setup logic located [here](https://github.com/WordPress/gutenberg/blob/trunk/packages/react-native-editor/src/setup.js). At this point, I'd agree that we might try to unify how the editor is initialized for tests, in order to prevent potential failures due to missing parts of the initialization logic.

> Another way of asking what I'm trying to understand is "if we suddenly want to make sure that the dispatch to the preferences store takes place when initializing an editor, but the test using this function dependency injects the Editor component, do they simply not care about the dispatch to the preferences store?" They'd have to dispatch manually in that use-case, right?

That's right, when initializing the editor using a custom editor component, we'd have to assure that it's properly initialized using the edit-post `initializeEditor` function. In the case of [the test you referenced](https://github.com/WordPress/gutenberg/blob/bc9fbe43f9e058eab54639cff65c8d62928b5fac/packages/react-native-editor/src/test/index.test.js#L184-L199), this is already covered as it runs the setup code, which in turn, retrieve the editor component using that function:
https://github.com/WordPress/gutenberg/blob/bc9fbe43f9e058eab54639cff65c8d62928b5fac/packages/react-native-editor/src/setup.js#L136-L138

Reply:
Thanks for helping me understand a bit more. Given all of this, I don't think there's a reason to not move forward with merging this change.